drivers/soundwire/cadence_master.c

```diff
@@ -1276,10 +1278,21 @@ EXPORT_SYMBOL(cdns_set_sdw_stream);
 static struct sdw_cdns_pdi *cdns_find_pdi(struct sdw_cdns *cdns,
 					  unsigned int offset,
 					  unsigned int num,
-					  struct sdw_cdns_pdi *pdi)
+					  struct sdw_cdns_pdi *pdi,
+					  int link_id,
+					  int dai_id)
 {
 	int i;
 
+	/* Check if we can find assigned pdi first */
+	for (i = offset; i < num; i++) {
+		if (pdi[i].link_id == link_id && pdi[i].dai_id == dai_id) {
```

Comment:
@bardliao I don't think we need to use the link_id. The PDIs are already relative to the DAI being used. I will take this patch and remove the link_id